tests/test_effective_config.py

```diff
@@ -0,0 +1,199 @@
+            self._unvalidated_keys.remove(key)
+
+        self.validation_messages.extend(validation_messages)
+
+
+class TestEffectiveConfig(TestBase):
+    """
+    Class used for testing EffectiveConfig
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(TestEffectiveConfig, self).__init__(*args, **kwargs)
+        self.effective_config = None
+        self.config_section = None
+
+    def init_effective_config(self):
+        """
+        This method is executed before each unit test
+        """
+        self.effective_config = EffectiveConfig()
+        self.config_section = CustomConfigSection(CUSTOM_SECTION_NAME, self.effective_config)
+        # Fill config section with some values
+        for key, value in CUSTOM_SECTION_VALUES.items():
+            self.config_section[key] = value
+        self.effective_config[CUSTOM_SECTION_NAME] = self.config_section
+
+    def test_effective_config_validate(self):
+        self.init_effective_config()
+        validate_messages = self.effective_config.validate()
+        self.assertEqual(validate_messages, [])
+
+    def test_effective_config_is_valid(self):
+        self.init_effective_config()
+        self.effective_config.validate()
+        is_valid = self.effective_config.is_valid()
+        self.assertEqual(is_valid, True)
+
+    def test_effective_config_virt_sections(self):
+        self.init_effective_config()
+        self.effective_config.validate()
+        virt_sections = self.effective_config.virt_sections()
+        self.assertEqual(len(virt_sections), 1)
+        self.assertEqual(virt_sections[0][0], CUSTOM_SECTION_NAME)
+
+    def test_effective_config_is_value_default(self):
+        self.init_effective_config()
+        self.effective_config.validate()
+        result = self.effective_config.is_default(CUSTOM_SECTION_NAME, 'my_str')
+        self.assertEqual(result, False)
+        result = self.effective_config.is_default(CUSTOM_SECTION_NAME, 'another_str')
+        self.assertEqual(result, True)
+
+    def test_effective_config_items(self):
+        self.init_effective_config()
+        self.effective_config.validate()
+        for key, item in self.effective_config.items():
+            self.assertEqual(key, CUSTOM_SECTION_NAME)
+            self.assertEqual(type(item), CustomConfigSection)
+
+    def test_effective_config_del_item(self):
+        self.init_effective_config()
+        self.effective_config.validate()
+        self.assertEqual(len(self.effective_config), 1)
+        del self.effective_config[CUSTOM_SECTION_NAME]
+        self.assertEqual(len(self.effective_config), 0)
+
+    def test_effective_config_filter_params(self):
+        effective_config = EffectiveConfig()
+        desired_params = [
+            'foo',
+            'bar',
+            'foo.bar'
+        ]
+        values = {
+            'foo': 'Foo',
+            'bar': 'Bar',
+            'bar.foo': 'Bar Foo',
+            'pub': 'Pub Foo'
+        }
```

Comment:
'Foo', 'Bar', 'Pub' ... the developer implementing this should be more original next time ;-).